core/src/Kokkos_Array.hpp

```diff
@@ -199,7 +199,7 @@ struct Array<T, KOKKOS_INVALID_INDEX, Array<>::contiguous> {
   using const_pointer   = std::add_const_t<T>*;
 
   KOKKOS_INLINE_FUNCTION constexpr size_type size() const { return m_size; }
-  KOKKOS_INLINE_FUNCTION constexpr bool empty() const { return 0 != m_size; }
+  KOKKOS_INLINE_FUNCTION constexpr bool empty() const { return 0 == m_size; }
```

Comment:
I'm surprised that this didn't cause any problem elsewhere. Was it just never used or never tested before?

Reply:
Definitely never tested before.

I suspect no one uses `Kokkos::Array<T, KOKKOS_INVALID_INDEX, Proxy>` directly.  As I state in Issue #6355, it has very unusual semantics (unlike `Kokkos::Array<T, N>`) as it is non-owning with weird (and untested until now) semantics for assignment.